--- D:/projects/ATCD/ACE/ace/WIN32_Asynch_IO.cpp	Sun Oct 08 12:58:10 2017
+++ C:/Users/Erik/Desktop/WIN32_Asynch_IO.cpp	Sun Oct 08 12:57:55 2017
@@ -168,25 +168,25 @@
 int
 ACE_WIN32_Asynch_Operation::cancel (void)
 {
-#if defined (ACE_HAS_WIN32_OVERLAPPED_IO)
-  // All I/O operations that are canceled will complete with the error
-  // ERROR_OPERATION_ABORTED. All completion notifications for the I/O
-  // operations will occur normally.
-
-  // @@ This API returns 0 on failure. So, I am returning -1 in that
-  //    case. Is that right? (Alex).
-  int const result = (int) ::CancelIo (this->handle_);
-
-  if (result == 0)
-    // Couldn't cancel the operations.
-    return 2;
-
-  // result is non-zero. All the operations are cancelled then.
-  return 0;
-
-#else /* !ACE_HAS_WIN32_OVERLAPPED_IO */
-  ACE_NOTSUP_RETURN (-1);
-#endif /* ACE_HAS_AIO_CALLS */
+#if defined (ACE_HAS_CANCEL_IO)
+  // All I/O operations that are canceled will complete with the error
+  // ERROR_OPERATION_ABORTED. All completion notifications for the I/O
+  // operations will occur normally.
+
+#if (_WIN32_WINNT < _WIN32_WINNT_VISTA)
+  int const result = (int) ::CancelIo (this->handle_);
+#else
+  int const result = (int) ::CancelIoEx (this->handle_, NULL);
+#endif
+  if (result == 0)
+    // Couldn't cancel the operations.
+    return 2;
+
+  // result is non-zero. All the operations are cancelled then.
+  return 0;
+#else /* !ACE_HAS_CANCEL_IO */
+  ACE_NOTSUP_RETURN (-1);
+#endif /* ACE_HAS_CANCEL_IO */
 }
 
 ACE_Proactor *
